src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -208,24 +212,29 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
    * First queries for the existence of the auth file, then deletes it if it exists.
    * If either of these operations fails, then a Future.failure is returned
    */
-  override def cleanUpForWorkflow(workflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = authenticated { connection =>
-    val gcsAuthFile = GoogleCloudStoragePath(gcsAuthFilePath(workflow))
-    val log = workflowLogger(workflow)
-    def gcsCheckAuthFileExists(prior: Option[Boolean]): Boolean = connection.storage.exists(gcsAuthFile)
-    def gcsAttemptToDeleteObject(prior: Option[Unit]): Unit = connection.storage.deleteObject(gcsAuthFile)
-    withRetry(gcsCheckAuthFileExists, log, s"Failed to query for auth file: $gcsAuthFile") match {
-      case Success(exists) if exists =>
-        withRetry(gcsAttemptToDeleteObject, log, s"Failed to delete auth file: $gcsAuthFile") match {
-          case Success(_) => Future.successful(Unit)
-          case Failure(ex) =>
-            log.error(s"Could not delete the auth file $gcsAuthFile", ex)
-            Future.failed(ex)
-        }
-      case Failure(ex) =>
-        log.error(s"Could not query for the existence of the auth file $gcsAuthFile", ex)
-        Future.failed(ex)
-      case _ => Future.successful(Unit)
-    }
+  override def cleanUpForWorkflow(workflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = {
+    Future(gcsAuthFilePath(workflow)) map { path =>
+      deleteAuthFile(path, workflowLogger(workflow))
+      ()
+    } recover { case _ => } // No need to fail here, it just means that we didn't have an auth file in the first place so no need to delete it.
```

Comment:
That's one reason the `Future` might have failed, but potentially there might be other less benign reasons.  Could this be more selective on the `Throwable` type?